### ISSUE TITLE
How to apply NCF to datasets that only have the number of interactions?

### ISSUE THREAD

**Issue by sgaseretto** [LaceyChen17/neural-collaborative-filtering#4]
As the question states, how could this be applied to a dataset that only has the number of interactions between an user and the item? Movielens has the ratings in the, which is explicit feedback, but how could this model be applied to a dataset like the [audioscrobbler dataset](https://github.com/sgaseretto/cf-spark/tree/master/data/audioscrobbling) which has as implicit-feedback the number of times a user heard an artist? Here is an example of recommendations implementing ALS and using that dataset: http://www.gousios.gr/courses/bigdata/audioscrobbler.html

**Comment by LaceyChen17** [comment]
I think NCF addresses implicit feedback as well. For dataset only containing interactions between users and items, you can try to use BPR(Bayesian Personalized Ranking) criterion as the loss function. Specifically, the existing interactions are positive samples while the negative ones are sampled manually.

**Comment by RuihongQiu** [comment]
According to the origin paper, NCF is proposed to deal with implicit feedback. May I ask why this repo used normalization to process ratings?

**Comment by LaceyChen17** [comment]
Negative items get 0 ratings in this repo. And I normalized the ratings into [0, 1]. I think it is fine if you do not normalize the rating. But it might be hard to tune the hyper-params then.

**Comment by RuihongQiu** [comment]
> Negative items get 0 ratings in this repo. And I normalized the ratings into [0, 1]. I think it is fine if you do not normalize the rating. But it might be hard to tune the hyper-params then.

I take a look at the implementation by the authors and found the function [here](https://github.com/hexiangnan/neural_collaborative_filtering/blob/4aab159e81c44b062c091bdaed0ab54ac632371f/NeuMF.py#L135). They just set all negative items to 0 and all other items interacted with users to 1 rather normalizing them (implicit feedback V.S. explicit feedback).

**Comment by LaceyChen17** [comment]
@RuihongQiu Thank you for reporting the bug. I added support for implicit feedback in the latest commit.  Could you check if it works well? I only tested the `GMF` .

**Comment by RuihongQiu** [comment]
@LaceyChen17 I will check it out soon.

**Comment by LaceyChen17** [comment]
Pull requests are extremely welcome!!! BTW Could you also share your training curves by updating README.md ?

**Comment by RuihongQiu** [comment]
Great! I will do it in one or two days.

**Comment by LaceyChen17** [comment]
Merged. Thank you so much!
